src/components/visible-while-frozen.js

```diff
@@ -36,10 +36,18 @@ AFRAME.registerComponent("visible-while-frozen", {
     const isFrozen = this.el.sceneEl.is("frozen");
 
     let isWithinDistance = true;
+    const isVisible = this.el.getAttribute("visible");
 
     if (this.data.withinDistance !== undefined) {
       getLastWorldPosition(this.cam, this.camWorldPos);
       this.objWorldPos.copy(this.el.object3D.position);
+
+      if (!isVisible) {
```

Comment:
Would probably move this above the `this.objWorldPos.copy` just so it reads a bit more clearly.